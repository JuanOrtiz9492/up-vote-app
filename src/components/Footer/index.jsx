import React from 'react';
import styled from 'styled-components';
import { FacebookIcon, TwitterIcon } from '../Icon/IconList';
import { cssConstants } from '../../config/cssConstants';
import { constants } from '../../config/constants';

const { coal } = cssConstants.colors;

const StyledFooter = styled.footer`
    display: flex;
    flex-direction: row;
    border-top: dotted 2px ${coal.hex};
    width: 100%;
    justify-content: space-between;
    padding: 1.5em 0;
    box-sizing: border-box;
`
const StyledLink = styled.a`
    text-decoration: none;
    color: ${coal.hex};
    margin-right: 2em;
`
const LinksContainer = styled.div`
    display: flex;
    flex-direction: row;
    height: fit-content;
    align-self: center;
    font-size: ${cssConstants.textSizes.small};
`
const TextContainer = styled.span`
    margin: 0 1em;
    height: fit-content;
    align-self: center;
    color: ${coal.hex};
`
const IconContainer = styled.div`
    margin-left: 2em;
    height: fit-content;
    align-self: center;
`
const Footer = ()=>(
    <StyledFooter>
        <LinksContainer>
            {constants.footer.moreInfo.map( item => (
                <StyledLink key={item.id} href={item.link}>{item.text}</StyledLink>
            ))}
        </LinksContainer>
        <LinksContainer>
            <TextContainer>Follow Us</TextContainer>
            <IconContainer>
                <FacebookIcon width={2} color={coal.hex}/>
            </IconContainer>
            <IconContainer>
                <TwitterIcon width={2} color={coal.hex}/>
            </IconContainer>
        </LinksContainer>
    </StyledFooter>
);

export default Footer;