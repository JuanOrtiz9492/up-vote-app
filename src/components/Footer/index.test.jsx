import React from 'react';
import { render } from '@testing-library/react';
import Footer from '.';

describe('Footer Component', () => {

    it('should Match the Snapshot', () => {
        const { container } = render(<Footer />);
        expect(container).toMatchSnapshot();
      })
})

