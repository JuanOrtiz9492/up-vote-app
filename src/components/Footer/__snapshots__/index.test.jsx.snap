// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer Component should Match the Snapshot 1`] = `
<div>
  <footer
    class="sc-AykKI jAHKFH"
  >
    <div
      class="sc-AykKK gZdhWe"
    >
      <a
        class="sc-AykKJ czMUkW"
        href="#"
      >
        Terms and Conditions
      </a>
      <a
        class="sc-AykKJ czMUkW"
        href="#"
      >
        Privacy Policy
      </a>
      <a
        class="sc-AykKJ czMUkW"
        href="#"
      >
        Contact Us
      </a>
    </div>
    <div
      class="sc-AykKK gZdhWe"
    >
      <span
        class="sc-fzXfLO kbaGEF"
      >
        Follow Us
      </span>
      <div
        class="sc-fzXfLP jFWvwJ"
      >
        <svg
          class="sc-AykKG dSASEA"
          color="#333333"
          width="2"
        >
          facebook-square.svg
        </svg>
      </div>
      <div
        class="sc-fzXfLP jFWvwJ"
      >
        <svg
          class="sc-AykKH jmtspN"
          color="#333333"
          width="2"
        >
          twitter.svg
        </svg>
      </div>
    </div>
  </footer>
</div>
`;
